src/progress_bars/ProgressBars.test.tsx

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import {ProgressBar, ProgressBarProps, ProgressBarType, ProgressBarStatus, ProgressBarAnimation} from ".";
+import {shallow, mount, ShallowWrapper} from "enzyme";
+import * as React from "react";
+
+let wrapper: ShallowWrapper<ProgressBarProps>;
+
+describe("<ProgressBar /> rendering", () => {
+    beforeAll(() => {
+        wrapper = shallow<ProgressBar>(<ProgressBar value={10} max={100} style={{width: "50%"}} />);
+    });
+
+    it("renders correctly", () => {
+        expect(wrapper).toMatchSnapshot();
+    });
+});
+
+describe("<ProgressBar /> rendering with label", () => {
+    beforeAll(() => {
+        wrapper = shallow<ProgressBar>(<ProgressBar value={10} max={100} labeled={true} style={{width: "50%"}} />);
+    });
+
+    it("renders correctly", () => {
+        expect(wrapper).toMatchSnapshot();
```

Comment:
For 'ProgressBar with label'  we can also add one more check that the label after progress bar and 'value' props to  ProgressBar should be same.

Reply:
This is inherently addressed as part of the snapshot verification.

Reply:
Can you please explain how snapshot will compare values of two HTML elements ?
I guess snapshot will only test the DOM structure of element. Correct me if I am wrong.

Reply:
Snapshot tests will compare complete html, but shallow rendered, so if next time same test is run with different props it will fail. Or if you accidentally make any changes in the component code it will fail.
If changes are intentional then we need to modify snapshot, if not then we need to fix the code.
So in short, we are taking care of everything being rendered here.

Reply:
Dropping this comment for now.

Reply:
@kulakshay So, if I pass value = {50} and then in case of labled progress bar the label value should be 50%. Are you saying that snapshot will automatically take care of comparing these values as well ?
As per my understanding snapshot testing is based on first snapshot that we create and then compare next snapshot with first one if we change anything in code.
If in next code suppose someone mistakenly assign label a hard-coded value of say 60% but value from props is 50. Does snapshot testing will catch such error?

Can't  we do something like below : refer code of MainContainer.test.tsx 
 it("should render the title", () => {
        expect(wrapper.find(".labeled").text()).toEqual("10");
    });

Reply:
> As per my understanding snapshot testing is based on first snapshot that we create and then compare next snapshot with first one if we change anything in code.

Yes that is correct

> If in next code suppose someone mistakenly assign label a hard-coded value of say 60% but value from props is 50. Does snapshot testing will catch such error?

It should, let's try it out

```
it("should render the title", () => {
expect(wrapper.find(".labeled").text()).toEqual("10");
});
```
For such testing we can not use snapshot testing, for this we need to render the component and find the label and compare it.

Reply:
Exactly, @akate1 raises a good point here in that snapshot testing will **not** identify if the label is correctly rendering what's passed in the `value` prop.  @payald, I don't think it will take long, can you add in this test?  You can check MainContainer.test.tsx for an example.  Thanks for your comments @kulakshay!

Reply:
Addressed comment